src/index.ts

```diff
@@ -209,4 +211,77 @@ export class SilentPayment {
   static addressToPubkey(address: string): string {
     return uint8ArrayToHex(bitcoin.address.toOutputScript(address).subarray(2));
   }
+
+  static getPubkeysFromTransactionInputs(tx: Transaction): Uint8Array[] {
+    const result: Uint8Array[] = [];
+
+    const stackToPubkeys = (stack: Stack): Uint8Array[] => {
+      return stack
+        .filter((elem) => typeof elem !== "number") // filtering out numbers, leaving only Uint8Array
+        .filter((elem) => ecc.isXOnlyPoint(elem as Uint8Array)) as Uint8Array[];
+    };
+
+    for (const input of tx.ins) {
+      const inScript = script.decompile(input.script);
+      if (inScript) {
+        // push any pubkeys in the scriptSig
+        result.push(...stackToPubkeys(inScript));
+        if (inScript.length > 1) {
+          const lastItem = inScript[inScript.length - 1];
+          if (typeof lastItem !== "number") {
+            // If the last item is a buffer, treat as redeemScript and check if we can decompile
+            // and if it has any pubkeys (it might not)
+            const redeemScript = script.decompile(lastItem);
+            if (redeemScript) {
+              result.push(...stackToPubkeys(redeemScript));
+            }
+          }
+        }
+      }
+      // Find any raw pubkeys in the witness stack
+      result.push(...input.witness.filter(script.isCanonicalPubKey));
+      for (const item of input.witness) {
+        const maybeScript = script.decompile(item);
+        if (maybeScript) {
+          result.push(...stackToPubkeys(maybeScript));
+        }
+      }
+    }
+    return result;
+  }
+
+  static computeTweakForTx(tx: Transaction): Uint8Array {
+    // you need the sum of the (eligible) input public keys (call it A), multiplied by the input_hash, i.e,
+    // hash(A|smallest_outpoint). this is a public key (33bytes) so this 33 bytes per tx is sent to the client.
+    // that would be a tweak (per tx)
+    const A = SilentPayment.sumPubKeys(SilentPayment.getPubkeysFromTransactionInputs(tx));
+
+    // looking for smallest outpoint:
+    const outpoints: Array<Uint8Array> = [];
+    for (const inn of tx.ins) {
+      const txidBuffer = inn.hash;//.reverse();
+      const voutBuffer = new Uint8Array(SilentPayment._ser32(inn.index).reverse());
+      outpoints.push(new Uint8Array([...txidBuffer, ...voutBuffer]));
+    }
+    outpoints.sort((a, b) => compareUint8Arrays(a, b));
+    const smallest_outpoint = outpoints[0];
+    const input_hash = SilentPayment.taggedHash("BIP0352/Inputs", concatUint8Arrays([smallest_outpoint, A]));
+
+    // finally, computing tweak:
+    return ecc.pointMultiply(A, input_hash);
+  }
+
+  static sumPubKeys(pubkeys: Uint8Array[], compressed: boolean = true): Uint8Array | null {
+    if (pubkeys.length === 0) return null;
+    if (pubkeys.length === 1) return concatUint8Arrays([new Uint8Array([2]), pubkeys[0]]); // not sure about this `2`
```

Comment:
@junderw i had to add leading `02` here, i guess it has smth to do with compressed pubkeys..? just want to make sure im not missing anything (my computations now match other implementation)

Reply:
changed to:

`.filter((elem) => ecc.isXOnlyPoint(elem as Uint8Array) || script.isCanonicalPubKey(elem as Uint8Array)) as Uint8Array[];`